src/main/java/de/tum/in/www1/artemis/config/ProgrammingLanguageConfiguration.java

```diff
@@ -0,0 +1,153 @@
+        checkImageForAllProgrammingLanguagesDefined(languageSpecificBuildImages);
+        images = languageSpecificBuildImages;
+    }
+
+    /**
+     * Parses the configuration contents into a properly typed map.
+     *
+     * @param imageConfig the config as it was received on server startup.
+     * @return A map of programming languages to the CI build images that are defined for it.
+     */
+    private Map<ProgrammingLanguage, Map<ProjectType, String>> loadImages(final Map<String, Map<String, String>> imageConfig) {
+        final EnumMap<ProgrammingLanguage, Map<ProjectType, String>> buildImages = new EnumMap<>(ProgrammingLanguage.class);
+
+        for (var entry : imageConfig.entrySet()) {
+            final ProgrammingLanguage programmingLanguage = tryParseProgrammingLanguage(entry.getKey());
+            final Map<ProjectType, String> projectTypeImages = tryParseProjectTypeBuildImages(entry.getValue());
+
+            buildImages.put(programmingLanguage, projectTypeImages);
+        }
+
+        return buildImages;
+    }
+
+    /**
+     * Tries to parse the given programming language.
+     * <p>
+     * Aborts server startup by throwing an exception if the language cannot be parsed.
+     *
+     * @param language The name of the language that should be parsed.
+     * @return The given language, now parsed.
+     */
+    private ProgrammingLanguage tryParseProgrammingLanguage(final String language) {
+        try {
+            return ProgrammingLanguage.valueOf(language.toUpperCase(Locale.ROOT));
+        }
+        catch (IllegalArgumentException ex) {
+            throw new IllegalArgumentException("Unknown programming language: " + language);
+        }
+    }
+
+    /**
+     * Tries to parse key-value pairs of project types and their build images.
+     * <p>
+     * Aborts server startup by throwing an exception if the project type cannot be parsed.
+     *
+     * @param buildImageConfig A map of project types to CI build images.
+     * @return A map of project types to CI build images.
+     */
+    private Map<ProjectType, String> tryParseProjectTypeBuildImages(final Map<String, String> buildImageConfig) {
+        final EnumMap<ProjectType, String> projectTypeImages = new EnumMap<>(ProjectType.class);
+
+        for (var entry : buildImageConfig.entrySet()) {
+            final String projectTypeRaw = entry.getKey();
+            final String image = entry.getValue();
+
+            final ProjectType projectType = switch (projectTypeRaw) {
+                case "default" -> DEFAULT_PROJECT_TYPE;
+                case "maven" -> MAVEN_PROJECT_TYPE;
+                case "gradle" -> GRADLE_PROJECT_TYPE;
+                case "gcc" -> ProjectType.GCC;
+                case "fact" -> ProjectType.FACT;
+                case "xcode" -> ProjectType.XCODE;
+                default -> throw new IllegalArgumentException("Unknown project type: " + projectTypeRaw);
```

Comment:
I think we should refactor that. Maybe move that to the enum. If another project type get's added we would have to search for switch cases.

Reply:
Good point. I moved it inside the `ProjectType` enum.